ipfs/service-worker-gateway
# fix: enforce origin isolation on subdomain gws

## Description

Towards #30: this PR detects when a path gateway URL is loaded  on host that supports subdomain gateway URLs, and redirects to unique origin.

To test:

1. Open http://helia-sw-gateway.localhost:3000 to register SW
1. Open http://helia-sw-gateway.localhost:3000/ipns/cid.ipfs.tech 
1. Page should redirect  to http://cid-ipfs-tech.ipns.helia-sw-gateway.localhost:3000

Caveats: 
- in Brave this works only with `helia-sw-gateway.localhost`, they seem to block `localhost`, so use `helia-sw-gateway.localhost` for testing
- check depends on CORS





## Notes & open questions



- [x] added follow-up tasks to  #30


## Change checklist

- [x] I have made corresponding changes to the documentation if necessary (this includes comments as well)
